File: January 2026 GFG SOLUTION/January-27.py
class Solution:
    def isWordExist(self, b, w):
        def search(r, c, idx):
            if idx == len(w): return True
            if r < 0 or c < 0 or r >= len(b) or c >= len(b[0]) or b[r][c] != w[idx]: return False
            tmp, b[r][c] = b[r][c], '*'
            res = search(r + 1, c, idx + 1) or search(r - 1, c, idx + 1) or search(r, c + 1, idx + 1) or search(r, c - 1, idx + 1)
            b[r][c] = tmp
            return res
        
        return any(search(i, j, 0) for i in range(len(b)) for j in range(len(b[0])))

File: January 2026 GFG SOLUTION/January-27.java
class Solution {
    public boolean isWordExist(char[][] b, String w) {
        for (int i = 0; i < b.length; i++)
            for (int j = 0; j < b[0].length; j++)
                if (search(b, w, i, j, 0)) return true;
        return false;
    }
    
    boolean search(char[][] b, String w, int r, int c, int idx) {
        if (idx == w.length()) return true;
        if (r < 0 || c < 0 || r >= b.length || c >= b[0].length || b[r][c] != w.charAt(idx)) return false;
        char tmp = b[r][c];
        b[r][c] = '*';
        boolean res = search(b, w, r + 1, c, idx + 1) || search(b, w, r - 1, c, idx + 1) ||
                      search(b, w, r, c + 1, idx + 1) || search(b, w, r, c - 1, idx + 1);
        b[r][c] = tmp;
        return res;
    }
}
